### PR TITLE
Upgrade docsearch to v3

### PR DESCRIPTION
**Pull request by pnbrown**
Updating docsearch to fix the search functionality on the site

### CODE REVIEW

**Comment by pnbrown** [comment]
Closes https://github.com/projectcontour/contour/issues/5086

**Comment by sunjayBhatia** [comment]
docs release note guidelines here: https://github.com/projectcontour/contour/blob/main/changelogs/unreleased/docs-sample.md

something simple like "Upgrade algolia docsearch to v3" should be good

**Comment by sunjayBhatia** [comment]
looks like it is working on the deploy preview: https://deploy-preview-5129--projectcontour.netlify.app/

**Comment by skriss** [comment]
> looks like it is working on the deploy preview: https://deploy-preview-5129--projectcontour.netlify.app/

Yep, preview looks great to me, the updated UI seems pretty nice!

File: changelogs/unreleased/5129-pnbrown-docs [hunk 1]
```diff
@@ -0,0 +1 @@
+Upgrade algolia docsearch to v3 on the docs website
```
Comment (skriss) [inline changelogs/unreleased/5129-pnbrown-docs:1]: Looks like you need to add a `.md` extension to the filename

Comment (pnbrown) [reply changelogs/unreleased/5129-pnbrown-docs]: Yup. Just caught that. My bad.

**Review by skriss — APPROVED** [review]
:+1:

**Review by sunjayBhatia — APPROVED** [review]

**Pull request merged**